PIVX-Labs/MyPIVXWallet
# Rewrite activity in vue

## Abstract

This is the first of a hopefully series of PRs aimed at rewriting parts MPW in Vue.js

## Why vue.js?

Vue is a fairly light framework, and can be integrated into our current codebase from the botton up, meaning that we can start by rewriting singular components, then entire tabs and maybe the entire page eventually.
The syntax is very similar to old school web development, but it avoids a lot of the boilerplate.
For instance, take this snippet of code, repeated many times in our codebase:
```js
document.getElementById('id').style.display = variable ? 'block' : 'none';
```
In vue, this becomes
```html
<div v-if="variable"> </div>
```
Note that we no longer need to assign an id to the div.
Other benefits include:
- The ability to split code in multiple files (index.template.html no longer needs to include everything).
- Harder to include vulnerabilities such as Xss
- Ability to containerize css (and js) to the component, and avoiding having big namespaces
- Probably more?

---

## Why start with the activity?

The activity list is a fairly complex component, which is reused two times (Dashboard and stake tab). It is a perfect example on how other parts can be rewritten as it needs to fetch transaction from the network, process the information and display it.

---

## Testing
- Test that the activity and stake reward list matches with [app.mypivxwallet.org](https://app.mypivxwallet.org/).
- Check that both of the lists are changed upon translation.
- Check that both of the lists update properly when Translations are changed.

---
